src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -275,6 +304,34 @@ public PriorityQueue<T> search(float[] queryVector, int limit, float threshold,
         return keyQueue;
     }
 
+    /**
+     * @return keys (PrimaryKey or segment row id) associated with the topK vectors near the query
+     */
+    public Pair<SearchResult, Supplier<SearchResult>> searchTopK(float[] queryVector, int limit, Bits toAccept)
```

Comment:
can we extract the common code from search+searchTopK, or have the former call the latter?